tui2/pages/output/output.go

```diff
@@ -75,7 +76,7 @@ func New(c common.Common, config *request.Config) (*Output, error) {
 	getBytesEncodingPerNetwork := func(endpoint string) dynamic.BytesRepresentation {
 		bytesAsBase58Chains := []string{"solana-mainnet-beta", "solana-mainnet", "solana-devnet"}
 		for _, chain := range bytesAsBase58Chains {
-			if manifest.HardcodedEndpoints[chain] == endpoint {
+			if networks.GetRegistryNetworksWithSubstreams().Find(chain).Services.Substreams[0] == endpoint {
 				return dynamic.BytesAsBase58
 			}
 		}
```

Comment:
The whole `getBytesEncodingPerNetwork` should be refactored actually while reviewing this. There is no good reason to check just for Solana here. The registry can determined network desired bytes representation via `Firehose.BytesEncoding`. 

Can you move that function to `networks` package `GetBytesRepresentation(key string)`. The `config.Endpoint` lookup is a bit weird also, because `config.Pkg.Network` is better to try first, but it's fair to then try by endpoint if `network`  is not defined and unfindable. Rules:

- Use existing network search rules
- If nothing found, make an endpoint search like now

Reply:
This code would then be refactored something like:

```
lookupKey := config.Pkg.Network
if lookupKey == "" { lookupKey = config.Endpoint }

bytesRepresentation := networks.GetBytesRepresentation(lookupKey)
```

Reply:
Ok, I didn't implement exactly how you said to try to keep network as "clean" as possible.
I added a `FindBySubstreamsEndpoint` method and `GetBytesEncoding` in `chainconfig.go`.
Then in `output.go` I try to find the network using `config.Pkg.Network` if it's set, else I try to find by endpoint. I then convert the result in `dynamic.BytesRepresentation` which is the type used in `output.go`.
It defaults to `hex` as before.